providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -778,10 +778,20 @@ int sm2_get_params(void *key, OSSL_PARAM params[])
         return 0;
 
     /* XXX:
-     * I dropped the support of OSSL_PKEY_PARAM_SECURITY_BITS since
+     * We assume SM2 security bits the same as in normal EC case since
      * I didn't find definition of SM2 security bits so far. This could
-     * be supported if the definition is clear in the future.
+     * be updated if the definition is clear in the future.
      */
+    if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_SECURITY_BITS)) != NULL) {
+        int ecbits, sec_bits;
+
+        ecbits = EC_GROUP_order_bits(ecg);
+        /* SM2 has only one curve so the sec_bits should always be a constant */
+        sec_bits = ecbits / 2;
```

Comment:
Is it suitable to hard-code the sec_bits instead of calculating?

Reply:
This had me look further, because I was sure that this was already done...  and yep, it is, in ec_set_params()!  So I wonder, why have an entirely different function for SM2 that does almost the exact same thing, when you could have a common set_params function that you pass a flag to indicate EC or SM2, if that's even necessary?

Having two functions with almost exactly the same code means that when something needs to be fixed, they must be kept in sync, which is a maintenance burden that is easily forgotten, especially when new coders come along and don't even notice there are copies.  Let's avoid that when we can!

Reply:
This is easy to achieve, actually I considered this last night but no enough time to do that but today I will go to fix this!

Reply:
Fixed!